### PR TITLE
Add Support for Kilo Plus Air Purifier

### PR DESCRIPTION
**Pull request by broyuken**
Adding support for kilo2 air purifier. Tested in my instance as working.

### CODE REVIEW

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    name: Timer
+    translation_key: timer
+    category: config
+    dps:
+      - id: 18
+        type: string
+        name: option
+        mapping:
+          - dps_val: cancle
+            value: "Off"
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:46]: updated.

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+name: Kilo Plus Air purifier
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:1]: updated.

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            value: "1 Hour"
+          - dps_val: "2h"
+            value: "2 Hours"
+          - dps_val: "3h"
+            value: "3 Hours"
+          - dps_val: "4h"
+            value: "4 Hours"
+  - entity: select
+    name: Light
+    translation_key: light
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:57]: updated

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          - dps_val: "2"
+            value: "On"
+  - entity: sensor
+    name: Countdown Remaining
+    category: diagnostic
+    dps:
+      - id: 20
+        type: integer
+        name: sensor
+        unit: "mins"
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:77]: updated

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        name: option
+        mapping:
+          - dps_val: cancle
+            value: "Off"
+          - dps_val: "1"
+            value: "Gradient"
+          - dps_val: "2"
+            value: "On"
+  - entity: sensor
+    name: Countdown Remaining
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:71]: updated.

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  - entity: switch
+    name: Reset Filter Life
+    category: config
+    icon: "mdi:toggle-switch-variant-off"
+    dps:
+      - id: 11
+        name: switch
+        type: boolean
+  - entity: select
+    name: Timer
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:37]: updated.

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    name: Filter
+    icon: "mdi:air-filter"
+    category: diagnostic
+    dps:
+      - id: 5
+        type: integer
+        name: sensor
+        unit: "%"
+  - entity: switch
+    name: Reset Filter Life
```
[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:29]: I removed the name and icon and added the translation_key, but the name shows as "None" now.

image

```
  - entity: switch
    translation_key: filter_reset
    category: config
    dps:
      - id: 11
        name: switch
        type: boolean
```

[the remark this answers is not in the document]
Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml]: Also tried changing it to a button based on the docs but that didn't change the name either. 

```
  - entity: button
    translation_key: filter_reset
    category: config
    dps:
      - id: 11
        name: button
        type: boolean
```

**Comment by broyuken** [comment]
What does this mean? Those files aren’t related to my PR

**Comment by broyuken** [comment]
This still doesn’t work correctly. It seems to pass your checks but the name doesn’t populate like you say it should. Please advise.

File: custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml [hunk 8]
```diff
@@ -1,4 +1,4 @@
-name: Air purifier
+name: Air Purifier
```
Comment (make-all) [inline custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:1]: This change is incorrect.

https://developers.home-assistant.io/blog/2022/07/10/entity_naming/

Comment (broyuken) [reply custom_components/tuya_local/devices/kilo_plus_air_purifier.yaml:1]: You are correct. Thanks for all this info, this is my first HA related PR.

**Pull request merged**